framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -0,0 +1,236 @@
+#include "LStableDirk4.h"
+
+// libMesh includes
+#include "libmesh/sparse_matrix.h"
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/preconditioner.h"
+
+registerMooseObject("MooseApp", ActuallyExplicitEuler);
+
+template <>
+InputParameters
+validParams<ActuallyExplicitEuler>()
+{
+  InputParameters params = validParams<TimeIntegrator>();
+
+  MooseEnum solve_type("consistent lumped lump_preconditioned", "consistent");
+
+  params.addParam<MooseEnum>(
+      "solve_type",
+      solve_type,
+      "The way to solve the system.  A 'consistent' solve uses the full mass matrix and actually "
+      "needs to use a linear solver to solve the problem.  'lumped' uses a lumped mass matrix with "
+      "a simple inversion - incredibly fast but may be less accurate.  'lump_preconditioned' uses "
+      "the lumped mass matrix as a preconditioner for the 'consistent' solve");
+
+  params.addClassDescription(
+      "Implementation of Explicit/Forward Euler without invoking any of the nonlinear solver");
+
+  return params;
+}
+
+/**
+ * Helper class to apply preconditioner
+ */
+class LumpedPreconditioner : public Preconditioner<Real>
+{
+public:
+  LumpedPreconditioner(const NumericVector<Real> & diag_inverse)
+    : Preconditioner(diag_inverse.comm()), _diag_inverse(diag_inverse)
+  {
+  }
+
+  virtual void init() override
+  {
+    // No more initialization needed here
+    _is_initialized = true;
+  }
+
+  virtual void apply(const NumericVector<Real> & x, NumericVector<Real> & y) override
+  {
+    y.pointwise_mult(_diag_inverse, x);
+  }
+
+protected:
+  /// The inverse of the diagonal of the lumped matrix
+  const NumericVector<Real> & _diag_inverse;
+};
+
+ActuallyExplicitEuler::ActuallyExplicitEuler(const InputParameters & parameters)
+  : TimeIntegrator(parameters),
+    MeshChangedInterface(parameters),
+    _solve_type(getParam<MooseEnum>("solve_type")),
+    _explicit_residual(_nl.addVector("explicit_residual", false, PARALLEL)),
+    _explicit_euler_update(_nl.addVector("explicit_euler_update", true, PARALLEL)),
+    _mass_matrix_diag(_nl.addVector("mass_matrix_diag", false, PARALLEL))
+{
+  _Ke_time_tag = _fe_problem.getMatrixTagID("TIME");
+
+  // Try to keep MOOSE from doing any nonlinear stuff
+  _fe_problem.solverParams()._type = Moose::ST_LINEAR;
+
+  if (_solve_type == LUMPED || _solve_type == LUMP_PRECONDITIONED)
+    _ones = &_nl.addVector("ones", false, PARALLEL);
+}
+
+void
+ActuallyExplicitEuler::initialSetup()
+{
+  meshChanged();
+}
+
+void
+ActuallyExplicitEuler::init()
+{
+}
+
+void
+ActuallyExplicitEuler::preSolve()
+{
+}
+
+void
+ActuallyExplicitEuler::computeTimeDerivatives()
+{
+  _u_dot = *_solution;
+  _u_dot -= _solution_old;
+  _u_dot *= 1 / _dt;
+  _u_dot.close();
+
+  _du_dot_du = 1.0 / _dt;
+}
+
+void
+ActuallyExplicitEuler::solve()
+{
+  // This ensures that all the Output objects in the OutputWarehouse
+  // have had solveSetup() called, and sets the default solver
+  // parameters for PETSc.
```

Comment:
Copy and paste from `LStableDirk4`. I do not see a value here. One either looks at the doxygen comment for this method or the actual implementation.

Also, this call is needed in multistep methods, because they involve multiple solves per time step. That's not really the case here. Did you try to remove this call? My guess is it is not really needed, `FEProblemBase::solve()` should call this for you...

Reply:
I don't use `FEProblemBase::solve()`... so I need to call this.

Reply:
As far as I can tell, the call stack is roughly like this right now:

* Transient::step()
* TimeStepper::step()
* FEProblemBase::solve()
* NonlinearSystem::solve()
* TimeStepper::solve()

Reply:
Hmm - that does seem right (other than the last line should be `TimeIntegrator::solve()` of course).

I suppose I can try to remove that call... let's see.

Reply:
Right, `TimeIntegrator`, my bad...